LabExT/View/MeasurementTable.py

```diff
@@ -114,9 +168,45 @@ def __setup__(self):
         self._tree.bind("<Motion>", self.handle_tooltip)
         self._tree.bind("<Leave>", self.hidetip)
 
+    def add_selection_changed_callback(self, callback: Callable[[SelectionChangedEvent], None]):
+        """Adds a selection changed listener to the list of listeners.
+
+        We model the listeners with a simple callback function which accepts a `SelectionChangedEvent`.
+
+        Args:
+            callback: The callable accepting a `SelectionChangedEvent` to add to the list of listeners
+        """
+        self._selection_changed_callbacks.append(callback)
+
+    def remove_selection_changed_callback(self, callback: Callable[[SelectionChangedEvent], None]):
+        """Removes one selection-changed-listener."""
+        self._selection_changed_callbacks.remove(callback)
+
+    def _notify_selection_changed_listeners(self, item_id_or_ids: Union[str, list[str]], is_checked: bool) -> None:
+        """Notifies all listeners about the newly selected item.
+
+        Args:
+            item_id: The id of the newly selected item in the `TreeView`
+            is_checked: True if and only if the item is now selected
+        """
+        checked_items = self._tree.get_checked()
+        for callback in self._selection_changed_callbacks:
+            if type(item_id_or_ids) == str:
+                measurements = self._hashes_of_meas[item_id_or_ids] if item_id_or_ids in self._hashes_of_meas else ""
+            else:
+                measurements = [self._hashes_of_meas[item] for item in item_id_or_ids if item in self._hashes_of_meas]
+            callback(
+                (
+                    item_id_or_ids,
+                    is_checked,
+                    [(hash, any([(hash in child) for child in checked_items])) for hash in self._hashes_of_meas.keys()],
+                    measurements,
+                )
```

Comment:
Could you please explain the function header you are using for the callback functions and its arguments?

Reply:
Yes, I think a comment would make sense here. thanks

Reply:
Done in 1ee47614120acd6a5498d43e9cedcaeee65b0d7a.